How-to-Build-a-Flask-ReactJS-Application-and-Deploy-to-Okteto.md

```diff
@@ -0,0 +1,647 @@
+Next, create a file for your API in `flask-react-app/api/todo_list_api.py`.
+
+
+Create `requirements.txt` file which you will use to list your dependencies. In your case you only have one and that is Flask.
+```
+Flask==1.1.2
+Flask-SQLAlchemy==2.5.0
+```
+[Flask-SQLAlchemy](https://flask-sqlalchemy.palletsprojects.com/en/2.x/) will be used ho handle the database in which the todos will stored.
+Now you are going to create a `Dockerfile`. This file will contain all the commands you want to call when building your image.
+
+```Dockerfile
+FROM python:3.8.2
+
+WORKDIR /api
+
+ENV FLASK_ENV=development
+
+COPY ./requirements.txt .
+
+RUN pip install -r requirements.txt
+
+COPY . .
+
+CMD [ "python", "todo_list_api.py" ]
+
+```
+Your file structure should now look like this:
+```
+├── flask-react-app
+│   ├── api
+│   │   ├── todo_list_api.py
+│   |   ├── Dockerfile
+│   |   ├── requirements.txt		
+
+```
+
+## Create-react-app
+You are going to create a react project using create-react-app.
+I am going to assume you have npm installed on your local device. That said, navigate to `flask-react-app` folder and run
+`$ npx create-react-app frontend`. 
+After it runs completely, some files will be created for you. 
+
+Since Okteto requires your application to start with port `8080` you will change it in your scripts. 
+In your text editor, navigate to `frontend/package.json`
+```json
+...
+"scripts": {
+	"start": "PORT=8080 react-scripts start",
+	"build": "react-scripts build",
+	"test": "react-scripts test",
+	"eject": "react-scripts eject"
+},
+...
+```
+Now you are going to create `Dockerfile` for your docker application.
+On your text editor, navigate to `frontend` folder and create `Dockerfile` and input the code:
+
+```Dockerfile
+ 
+FROM node:14.17.3
+
+WORKDIR /user/src/app 
+
+COPY ./package.json ./
+
+COPY ./package-lock.json ./
+
+RUN npm install 
+
+COPY . .
+
+EXPOSE 8080
+
+ENTRYPOINT [ "npm" ]
+
+CMD ["start"]
+
+```
+
+I know you would like to test your code, but let's first create your `docker-compose.yml` file so that you can run both servers at once.
+With the `docker-compose.yml` commands you will connect the images created for React and Flask and run them together. You will also use the `docker-compose.yml` to set how you would like Okteto to run your application.
+In `flask-react-app` folder create `docker-compose.yml` file and put the code:
+
+```yml
+services:
+  api:
+    build:  ./api
+    ports: 
+    - "5000:5000"
+    volumes: 
+    - ./api:/api
+    environment: 
+      FLASK_ENV: development
+  web:
+    build: ./frontend
+    ports: 
+      - "8080:8080"
+    volumes: 
+      - ./frontend:/user/src/app
+    depends_on: 
+      - api
+
+```
+Now run `$ docker-compose build` and then `$ docker-compose up`
+
+You'll see that your react application is running on port `localhost:8080`
+
+![](First-image)
```

Comment:
Can you provide an image of your frontend application running at localhost:8080?

Reply:
it's the default page that comes when you run `$ npx create-react-app`, which will be impossible to get unless I go through the steps from the beginning.